src/main/kotlin/com/campspot/lib/PunchLib.kt

```diff
@@ -4,20 +4,23 @@ import com.campspot.api.Punch
 import com.campspot.dao.PunchDAO
 import com.campspot.exceptions.EntityNotFoundException
 import com.campspot.exceptions.PunchCannotOverlapException
+import com.campspot.middleware.transactions.DAOManager
 import java.time.ZonedDateTime
 
 class PunchLib(
-  private val punchDAO: PunchDAO,
+  private val daoManager: DAOManager,
   private val mockableObject: MockableObject
 ) {
   fun create(punch: Punch): Punch {
+    val punchDAO = daoManager[PunchDAO::class]
```

Comment:
We can no longer use dao instances as properties because each instance needs to be tied to a thread/transaction. So you have to fetch it out of this container in each method